subprojects/execution/src/main/java/org/gradle/internal/execution/impl/OutputFilterUtil.java

```diff
@@ -85,8 +85,14 @@
      * Decide whether an entry should be considered to be part of the output. See class Javadoc for definition of what is considered output.
      */
     private static boolean isOutputEntry(Map<String, FileSystemLocationFingerprint> afterPreviousExecutionFingerprints, Map<String, CompleteFileSystemLocationSnapshot> beforeExecutionSnapshots, CompleteFileSystemLocationSnapshot afterExecutionSnapshot, Boolean isRoot) {
-        if (isRoot && afterExecutionSnapshot.getType() == FileType.Missing) {
-            return false;
+        if (isRoot) {
+            FileType afterExecutionSnapshotType = afterExecutionSnapshot.getType();
+            if (afterExecutionSnapshotType == FileType.Missing) {
+                return false;
+            }
+            if (afterExecutionSnapshotType == FileType.Directory) {
+                return true;
+            }
```

Comment:
Nit: I think this would be more obvious with a `switch`.

Reply:
Using a switch now.